pkg/functions/function.go

```diff
@@ -148,6 +148,14 @@ type BuildSpec struct {
 	// Image stores last built image name NOT in func.yaml, but instead
 	// in .func/built-image
 	Image string `yaml:"-"`
+
+	// Mounts used in build phase. This is useful in particular for paketo bindings.
+	Mounts []MountSpec `yaml:"mounts,omitempty"`
+}
+
+type MountSpec struct {
```

Comment:
Alternative to this new type would be to reuse our `Volume` type and just add `Source *string` member to it.
WDYT @lkingland ?

Reply:
It could to a degree help on cluster and local build unification.